src/usePromise.ts

```diff
@@ -1,75 +1,89 @@
-import { DependencyList, useEffect, useMemo, useState } from 'react'
+import { DependencyList, useEffect, useState } from 'react'
 
-export enum PromiseStatus {
-  Pending = 'pending',
-  Fulfilled = 'fulfilled',
-  Rejected = 'rejected',
-}
-
-export type PromiseResult<T> =
-  | PromisePendingResult
-  | PromiseFulfilledResult<T>
-  | PromiseRejectedResult
+export type PromiseResult<T> = PromisePendingResult | PromiseSettledResult<T>
 
 export interface PromisePendingResult {
-  status: PromiseStatus.Pending
```

Comment:
Why aren't we using enums anymore?

Reply:
Because of: https://github.com/Amsterdam/use-promise/pull/6#issuecomment-808939120